File: Serenity Practice/HosbitalSerenity.Web/Modules/ServerTypes/Hosbital/DoctorsForm.ts
import { StringEditor, LookupEditor, ImageUploadEditor, MultipleImageUploadEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";
import { DoctorPatientGrid } from "../../Hosbital/Doctors/DoctorPatientGrid";
import { SpecialityDialog } from "../../Hosbital/Speciality/SpecialityDialog";

export interface DoctorsForm {
    DoctorName: StringEditor;
    SpecialityList: LookupEditor;
    ProfilePicture: ImageUploadEditor;
    Documents: MultipleImageUploadEditor;
    PatientsGrid: DoctorPatientGrid;
}

export class DoctorsForm extends PrefixedContext {
    static readonly formKey = 'Hosbital.Doctors';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!DoctorsForm.init)  {
            DoctorsForm.init = true;

            var w0 = StringEditor;
            var w1 = LookupEditor;
            var w2 = ImageUploadEditor;
            var w3 = MultipleImageUploadEditor;
            var w4 = DoctorPatientGrid;

            initFormType(DoctorsForm, [
                'DoctorName', w0,
                'SpecialityList', w1,
                'ProfilePicture', w2,
                'Documents', w3,
                'PatientsGrid', w4
            ]);
        }
    }
}

queueMicrotask(() => [SpecialityDialog]); // referenced dialogs

File: Serenity Practice/HosbitalSerenity.Web/Modules/Hosbital/Doctors/DoctorsForm.cs
using Serenity.ComponentModel;

namespace HosbitalSerenity.Hosbital.Forms;

[FormScript("Hosbital.Doctors")]
[BasedOnRow(typeof(DoctorsRow), CheckNames = true)]
public class DoctorsForm
{
    [Tab("Doctors")]
    public string DoctorName { get; set; }
    //public int SpecialityId { get; set; }
    public List<int> SpecialityList { get; set; }
    public string ProfilePicture { get; set; }
    public string Documents { get; set; }

    [Tab("Patients"), IgnoreName, DoctorPatientGrid, LabelWidth("0")]
    public string PatientsGrid { get; set; }
}